### PR TITLE
ErrorHandler fix

### LINKED ISSUE #3571: Infinite recursion in ErrorHandler
#### I got 255 return code without any error message when the first test of a second suite execution was started.

I fixed that, pull request will be as soon as possible.
You can reproduce this with https://github.com/positronium/sf-two-suites-with-deptecations

```bash
$ codecept run -f
Codeception PHP Testing Framework v2.2.5
Powered by PHPUnit 4.8.24 by Sebastian Bergmann and contributors.

Unit Tests (7) -----------------------------------------------
Testing unit
✔ DbSnapshotCest: Self test (8.61s)
✔ DbSnapshotCest: Dummy (0.00s)
✔ ToArrayCest: To array test (1.70s)
✔ ToArrayCest: Build array via white list test (0.39s)
✔ ToArrayCest: Build array via black list test (0.45s)
✔ ToArrayCest: For comet updated test (0.42s)
✔ ToArrayCest: For comet on off line test (0.36s)
-------------------------------------------------------------

Api Tests (15) ---------------------------------------------
- LoginCest: Log in as tester no | 0
[255] user@dev-machine [dev $%>] $
```

### Details

* Codeception version: 2.2.5
* PHP Version: 5.6.24
* Operating System: Debian 7 wheezy
* Installation type: Composer
* Suite configuration: `unit` and `api` (functional) suite

### PR DESCRIPTION
**Pull request by positronium**
Call `registerDeprecationErrorHandler` method and `register_shutdown_function` on first SuiteEvent only.
Fixes #3571.

### CODE REVIEW

File: src/Codeception/Subscriber/ErrorHandler.php [hunk 1]
```diff
@@ -39,11 +44,16 @@ public function handle(SuiteEvent $e)
             $this->errorLevel = eval("return {$settings['error_level']};");
         }
         error_reporting($this->errorLevel);
+
+        if (self::$initialized) {
```
Comment (sergeyklay) [inline src/Codeception/Subscriber/ErrorHandler.php]: Please don't introduce new static properties if it's not necessary

Comment (DavertMik) [inline src/Codeception/Subscriber/ErrorHandler.php]: @sergeyklay do you have other ideas?

Comment (positronium) [inline src/Codeception/Subscriber/ErrorHandler.php]: `set_error_handler()` acts globally, and I think a flag related with it also must be "global".
I just fixed a work-stopping issue for myself and appreciate a more sophisticated solution.

Comment (DavertMik) [inline src/Codeception/Subscriber/ErrorHandler.php]: it was done by the same manner as `self::$stopped`. Can we make `$stopped` to be instance variable as well?
Won't it break anything?

**Comment by DavertMik** [comment]
Thanks, I will do additional improvements

**Pull request merged**